rpcserver/socket_writes.cc

```diff
@@ -0,0 +1,50 @@
+#include <iostream>
+#include <asio.hpp>
+#include <netinet/in.h>
+#include <google/protobuf/io/coded_stream.h>
+
+using asio::ip::tcp;
+
+namespace rpcserver {
+
+    /**
+     * Return whether an attempt to write val on socket is successful.
+     */
+    bool write_int32(tcp::socket& sock, uint32_t val) {
+        asio::error_code error;
+        size_t write_len = sock.write_some(asio::buffer(&val, 4), error);
```

Comment:
Should htonl be called first, as is done symmetrically in socket_reads:read_int32?